tests/utils/test_utils.go

```diff
@@ -288,7 +297,109 @@ func ChangeWDWithCallback(t *testing.T, newPath string) func() {
+	xrayManager, err := xray.CreateXrayServiceManager(configTests.XrDetails)
+	require.NoError(t, err)
+	ignoreRuleId, err := xrayManager.CreateIgnoreRule(xrayUtils.IgnoreRuleParams{
+		// expired in one day
+		Notes:         description,
+		ExpiresAt:     time.Now().AddDate(0, 0, 1),
+		IgnoreFilters: filters,
+	})
+	assert.NoError(t, err)
+	return func() {
+		assert.NoError(t, xrayManager.DeleteIgnoreRule(ignoreRuleId))
+	}
+}
+
+func CreateSecurityPolicy(t *testing.T, policyName string, rules ...xrayUtils.PolicyRule) (string, func()) {
+	xrayManager, err := xray.CreateXrayServiceManager(configTests.XrDetails)
+	require.NoError(t, err)
+	// Create new default security policy.
+	policyParams := xrayUtils.PolicyParams{
+		Name:  fmt.Sprintf("%s-%s", policyName, strconv.FormatInt(time.Now().Unix(), 10)),
+		Type:  xrayUtils.Security,
+		Rules: rules,
+	}
+	if !assert.NoError(t, xrayManager.CreatePolicy(policyParams)) {
+		return "", func() {}
+	}
+	return policyParams.Name, func() {
+		assert.NoError(t, xrayManager.DeletePolicy(policyParams.Name))
+	}
+}
+
+func CreateTestSecurityPolicy(t *testing.T, policyName string, severity xrayUtils.Severity, failBuild bool) (string, func()) {
+	return CreateSecurityPolicy(t, policyName,
+		xrayUtils.PolicyRule{
+			Name:     "sca_rule",
+			Criteria: *xrayUtils.CreateSeverityPolicyCriteria(severity),
+			Actions:  getBuildFailAction(failBuild),
+			Priority: 1,
+		},
+		xrayUtils.PolicyRule{
+			Name:     "exposers_rule",
+			Criteria: *xrayUtils.CreateExposuresPolicyCriteria(severity, true, true, true, true, false),
+			Actions:  getBuildFailAction(failBuild),
+			Priority: 2,
+		},
+		xrayUtils.PolicyRule{
+			Name:     "sast_rule",
+			Criteria: *xrayUtils.CreateSastPolicyCriteria(severity),
+			Actions:  getBuildFailAction(failBuild),
+			Priority: 3,
+		},
+	)
+}
+
+func getBuildFailAction(failBuild bool) *xrayUtils.PolicyAction {
+	if failBuild {
+		return &xrayUtils.PolicyAction{
+			FailBuild: clientUtils.Pointer(true),
+		}
+	}
+	return nil
+}
+
+func createTestWatch(t *testing.T, policyName, watchName string, assignParams func(watchParams xrayUtils.WatchParams) xrayUtils.WatchParams) (string, func()) {
+	xrayManager, err := xray.CreateXrayServiceManager(configTests.XrDetails)
+	require.NoError(t, err)
+	// Create new default watch.
+	watchParams := assignParams(xrayUtils.NewWatchParams())
+	watchParams.Name = fmt.Sprintf("%s-%s", watchName, strconv.FormatInt(time.Now().Unix(), 10))
+	watchParams.Active = true
+	// Assign the policy to the watch.
+	watchParams.Policies = []xrayUtils.AssignedPolicy{
+		{
+			Name: policyName,
+			Type: string(xrayUtils.Security),
+		},
+	}
+	assert.NoError(t, xrayManager.CreateWatch(watchParams))
+	return watchParams.Name, func() {
+		assert.NoError(t, xrayManager.DeleteWatch(watchParams.Name))
+	}
+}
+
+func CreateTestGitRepoWatch(t *testing.T, policyName, watchName string, gitResources ...string) (string, func()) {
```

Comment:
I think the name of the function should be different since you can create a watch with git repository or with build type.
Maby we should call it  CreateWatchForTests?